pkg/cli/admin/nodeimage/basenodeimagecommand.go

```diff
@@ -87,9 +89,48 @@ func (c *BaseNodeImageCommand) getNodeJoinerPullSpec(ctx context.Context) error
 	}
 
 	// Extract the baremetal-installer image pullspec, since it
-	// provide the node-joiner tool.
+	// provides the node-joiner tool.
+
+	// First attempt to get the installer image from the configMap created by installer. This will work in disconnected environments.
+	installerImagesConfigMap, err := c.Client.CoreV1().ConfigMaps("openshift-config").Get(ctx, "installer-images", metav1.GetOptions{})
+	if err == nil {
+		images := installerImagesConfigMap.Data["images.json"]
+		if len(images) > 0 {
+			imageMap := make(map[string]string)
+			err := yaml.Unmarshal([]byte(images), &imageMap)
+			if err == nil {
+				installer, exists := imageMap["installer"]
+				if exists {
+					// Found pullSpec from configMap
+					c.log("installer pullspec obtained from installer-images configMap %s", installer)
+					c.nodeJoinerImage = installer
+					return nil
+				}
+			}
+		}
+	} else if !kapierrors.IsNotFound(err) {
+		c.log("failure retrieving installer-images configMap %s", err.Error())
+	} else {
+		c.log("configMap containing installer-images is not available, trying to get image from registry")
+	}
+
+	// If cannot obtain installer image from configMap, get it from the releaseImage.
+	// Note that after OCP 4.20, this should not be necessary since the configMap was added to installer in 4.19.
 	opts := ocrelease.NewInfoOptions(c.IOStreams)
 	opts.SecurityOptions = c.SecurityOptions
+	opts.IDMSFile = c.getIdmsFile(ctx)
+	if opts.IDMSFile == "" {
+		opts.ICSPFile = c.getIcspFile(ctx)
```

Comment:
Clusters are allowed to have both ICSP and IDMS objects. Ideally we would build a single IDMSFile from both sources rather than separate files from each and picking one.

Reply:
Are they both supported **at the same time**? I know there can be ICSP's after an upgrade but I thought it was [recommended](https://docs.redhat.com/en/documentation/openshift_container_platform/4.14/html/images/image-configuration#images-configuration-registry-mirror_image-configuration) to convert that to an IDMS.

I know with ABI only IDMS are created in the cluster, since 4.17 I believe - https://issues.redhat.com//browse/MGMT-14159. 

We could combine the mirrors but I assume that would just create duplicates, I don't think the intent is to split the mirroring config across the 2 objects.

Reply:
> Are they both supported **at the same time**?

Looks like yes unless something has changed: https://issues.redhat.com/browse/OCPNODE-1771

From your doc link:

>  For new clusters, you can use IDMS, ITMS, and ICSP CRs objects as desired. However, using IDMS and ITMS is recommended.
>
> If you upgraded a cluster, any existing ICSP objects remain stable, and both IDMS and ICSP objects are supported.

Reply:
OK, I was under the assumption that the ICSP was removed after it was converted to IDMS. If they're both there I'll concatenate all of the mirrors into one IDMS.